azure-pipelines.yml

```diff
@@ -46,7 +46,6 @@ stages:
   - template: tools/templates/acrbuild.yml
 - stage: coverage
   displayName: 'Code Coverage'
-  condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
```

Comment:
Why do pull requests need code coverage, we already do tests...  Pull requests are left out to keep the runs as fast as possible, given some flakiness already in the tests.

Reply:
PR should ideally gate on required code coverage to be checked in. This guarantees minimum testing needed for the code be checked in.

Reply:
The code coverage step is not a test.  It just runs the tests again.  We should change testing the outcome to not regress from previous runs.  Just running the test again is not extra benefit, quite the opposite.

Reply:
Interesting, then Code coverage step needs change. Either it can be merged with Testing to generate report then or if extra step is needed we need to gate on minimum required tests to be checked in any code. I'll make these changes as next PR then. Too big a change for this PR.